matlab/+caffe/private/caffe_.cpp

```diff
@@ -45,7 +45,11 @@ void mxCHECK_FILE_EXIST(const char* file) {
 static vector<shared_ptr<Solver<float> > > solvers_;
 static vector<shared_ptr<Net<float> > > nets_;
 // init_key is generated at the beginning and everytime you call reset
-static double init_key = static_cast<double>(caffe_rng_rand());
```

Comment:
Can you use `_MSC_VER` to ensure that the modification is enabled only on Windows?